tools/osx/crosstool/cc_toolchain_config.bzl.tpl

```diff
@@ -5402,6 +5391,88 @@ def _impl(ctx):
         ],
     )
 
+    headerpad_feature = feature(
+        name = "headerpad",
+        enabled = True,
+        flag_sets = [
+            flag_set(
+                actions = all_link_actions + [
+                    ACTION_NAMES.objc_executable,
+                    ACTION_NAMES.objcpp_executable,
+                ],
+                flag_groups = [flag_group(flags = ["-headerpad_max_install_names"])],
+                with_features = [with_feature_set(not_features = [
+                    "bitcode_embedded",
+                    "bitcode_embedded_markers",
+                ])],
+            ),
+        ],
+    )
+
+    if (ctx.attr.cpu == "ios_arm64"
+        or ctx.attr.cpu == "ios_arm64e"
+        or ctx.attr.cpu == "ios_armv7"
+        or ctx.attr.cpu == "tvos_arm64"
```

Comment:
Do this list need to be regularly updated as new devices are released? For example if there's an update that adds tvos_arm64e?

Reply:
For now, yes, though we plan to refactor this a bit based on some of the experience we gained by porting our other CROSSTOOLs to Starlark.